### PR TITLE
Docathon: Add docs for nn.functional.*d_max_pool

### PR DESCRIPTION
**Pull request by Gamrix**
Adding docs to max_pool to resolve docathon issue #60904

Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63264

Differential Revision: [D31071491](https://our.internmc.facebook.com/intern/diff/D31071491)

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -606,10 +606,26 @@ def max_pool1d_with_indices(
...
+        kernel_size: the size of the window. Can be a single number or a
+            tuple `(kW,)`
+        stride: the stride of the window. Can be a single number or a tuple
+            `(sW,)`. Default: :attr:`kernel_size`
+        padding: Implicit negative infinity padding to be added on both sides, must be >= 0 and <= kernel_size / 2.
+        dilation: The stride between elements within a sliding window, must be > 0.
+        ceil_mode: If ``True``, will use `ceil` instead of `floor` to compute the output shape. This
+                   ensures that every element in the input tensor is covered by a sliding window.
+        return_indices: If ``True``, will return the argmax along with the max values.
+                        Useful for :class:`torch.nn.MaxUnpool1d` later
```
Comment (jbschlosser) [inline torch/nn/functional.py:628]: possibly reference the functional `max_unpool1d` instead? Same throughout

**Review by jbschlosser — COMMENTED** [review]
Thanks for the contribution! Left a few minor comments below :)

**Review by jbschlosser — APPROVED** [review]
LGTM thanks!

**Comment by Gamrix** [comment]
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31071491).

**Pull request closed without merging**